### PR TITLE
[TASK] Add documentation about FileProcessors

### PR DESCRIPTION
**Pull request by sabbelasichon**
Resolves: https://github.com/rectorphp/rector/issues/6374

### CODE REVIEW

**Comment by sabbelasichon** [comment]
@lulco Would love to hear your feedback. Thanks for your notes.

**Comment by sabbelasichon** [comment]
@TomasVotruba And of course yours too ;-)

Afterwards i am gonna document the new Formatters. They are somehow connected to this topic.

File: build/target-repository/docs/how_to_create_custom_fileprocessor.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Create your own custom FileProcessor
+
+This section is all about creating your custom specific FileProcessor.
+If you don´t know the concept of FileProcessors in the context of Rector, have a look at [Beyond PHP - Entering the realm of FileProcessors](beyond_php_file_processors.md)
+
+Most of the examples starting with a rather contrived example, let´s do it the same.
+
+Imagine you would like to replace the sentence "Make america great again" to "Make the whole world a better place to be" in every file named dumb_trump.txt.
```
Comment (lulco) [inline build/target-repository/docs/how_to_create_custom_fileprocessor.md]: Not sure if I would use Trump or any politician in examples :) (not Trump's fan)

File: build/target-repository/docs/how_to_create_custom_fileprocessor.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    public function transform(File $file): void
+    {
+        $oldContent = $file->getFileContent();
+
+        if(false === strpos($oldContent, self::OLD_STATEMENT)) {
+            return;
+        }
+
+        $newFileContent = str_replace(self::OLD_STATEMENT, 'Make the whole world a better place to be', $oldContent);
+        $file->changeFileContent($newFileContent);
```
Comment (lulco) [inline build/target-repository/docs/how_to_create_custom_fileprocessor.md]: this should be called only in FileProcessor not in rectors

File: build/target-repository/docs/how_to_create_custom_fileprocessor.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+use Rector\Core\ValueObject\Application\File;
+
+final class DumpTrumpMakeTheWorldGreatAgainRector implements DumpTrumpRectorInterface
+{
+    /**
+    * @var string
+    */
+    private const OLD_STATEMENT = 'Make america great again';
+
+    public function transform(File $file): void
```
Comment (lulco) [inline build/target-repository/docs/how_to_create_custom_fileprocessor.md]: I prefer content as parameter, not whole File

File: build/target-repository/docs/how_to_create_custom_fileprocessor.md (hunk 1 above)
Comment (lulco) [reply build/target-repository/docs/how_to_create_custom_fileprocessor.md]: @sabbelasichon have you pushed changes? I still see DumpTrump and only one commit for this PR

**Comment by sabbelasichon** [comment]
@lulco  I have updated it now. Didn´t push it. Sorry.

**Comment by lulco** [comment]
👍

**Comment by sabbelasichon** [comment]
@TomasVotruba Do you think it is a good start?

**Comment by TomasVotruba** [comment]
Looks good to me :+1:

**Comment by TomasVotruba** [comment]
Let's ship it then :slightly_smiling_face:  :+1:

**Pull request merged**